ledger/appcow.go

```diff
@@ -602,16 +607,18 @@ func applyStorageDelta(data basics.AccountData, aapp storagePtr, store *storageD
 		case deallocAction:
 			delete(owned, aapp.aidx)
 		case allocAction, remainAllocAction:
-			// TODO verify this assertion
 			// note: these should always exist because they were
 			// at least preceded by a call to Put?
 			states, ok := owned[aapp.aidx]
 			if !ok {
 				return basics.AccountData{}, fmt.Errorf("could not find existing states for %v", aapp.aidx)
 			}
 			states = states.Clone()
-			if store.action == allocAction {
-				// TODO does this ever accidentally clobber?
+			if store.action == allocAction ||
+				store.action == remainAllocAction && states.KeyValue == nil {
```

Comment:
it's true that in go, `&&` operator has a higher precedence compared to `||`. But don't force the reader of that code to know that. Maybe use round brackets ?